### PR TITLE
WebGLShadowmap: Handle `TwoPassDoubleSide` case

### PR DESCRIPTION
**Pull request by WestLangley**

### CODE REVIEW

**Comment by WestLangley** [comment]
@gkjohnson @Mugen87 

In light of this PR, does this code block need to be changed?

https://github.com/mrdoob/three.js/blob/f296f21c6bb87ec382af402c8685724a88eddc4c/src/renderers/webgl/WebGLShadowMap.js#L280-L288

**Comment by gkjohnson** [comment]
With a quick look - it seems like it should be okay? The renderer falls back to using DoubleSided renderering with non transparent materials. But maybe it's worth being explicit?

https://github.com/mrdoob/three.js/blob/87223b87ed93da282f75a27df06c7ca7e39437d0/src/renderers/WebGLRenderer.js#L831-L847

File: src/renderers/webgl/WebGLShadowMap.js [hunk 1]
```diff
@@ -28,7 +28,7 @@ function WebGLShadowMap( _renderer, _objects, _capabilities ) {
 
 		_maxTextureSize = _capabilities.maxTextureSize;
 
-	const shadowSide = { 0: BackSide, 1: FrontSide, 2: DoubleSide };
+	const shadowSide = { 0: BackSide, 1: FrontSide, 2: DoubleSide, 3: DoubleSide };
```
Comment (LeviPesin) [inline src/renderers/webgl/WebGLShadowMap.js:31]: Shouldn't we do `{ [ FrontSide ]: BackSide, [ BackSide ]: FrontSide, ... }` here?

**Review by Mugen87 — APPROVED** [review]

**Comment by WestLangley** [comment]
Rephrasing my question... Is VSM handled correctly given `material.side` can now be `TwoPassDoubleSide`? (Should it map to `DoubleSide`?)

https://github.com/mrdoob/three.js/blob/f296f21c6bb87ec382af402c8685724a88eddc4c/src/renderers/webgl/WebGLShadowMap.js#L280-L288

FYI, see [related comment](https://github.com/mrdoob/three.js/pull/17171#pullrequestreview-270942491).

**Comment by gkjohnson** [comment]
> Rephrasing my question... Is VSM handled correctly given material.side can now be TwoPassDoubleSide? (Should it map to DoubleSide?)

What I'm saying is that the render behavior for shadows with `DoubleSide` and `TwoPassDoubleSide` should be the same since the materials are not transparent.

**Comment by WestLangley** [comment]
Thanks, @gkjohnson for your reply.

I still am not a fan this design, but the code needs to be fixed, so merging.

**Pull request merged**